src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseStream.cs

```diff
@@ -17,6 +17,7 @@ internal class WinHttpResponseStream : Stream
         private volatile bool _disposed;
         private readonly WinHttpRequestState _state;
         private SafeWinHttpHandle _requestHandle;
+        private CancellationTokenRegistration _ctr;
```

Comment:
I explicitly did not do that because those TaskCompletionSource's are completed in the WinHttpRequestCallback class and I didn't want to have to reach into the WinHttpResponseStream itself. In fact, once the callback itself doesn't have any pointer back to the current stream. The only thing it knows about is the current TcsReadFromResponseStream that is stored in the _state (WinHttpRequestState).

So, I was basically letting the GC take care of this especially for repeated reads.